tests/integration/test_utils_vcs_git.py

```diff
@@ -39,6 +40,15 @@
 pytestmark = pytest.mark.integration
 
 
+class GitCloneKwargs(TypedDict):
```

Comment:
Kind of feel this was over kill, but seems this was the only way to stop mypy complaining. Suggestions welcome. 

```console
tests/integration/test_utils_vcs_git.py:149: error: Argument 2 to "clone" of "Git" has incompatible type "**dict[str, str]"; expected "Path | None"  [arg-type]
tests/integration/test_utils_vcs_git.py:149: error: Argument 2 to "clone" of "Git" has incompatible type "**dict[str, str]"; expected "bool"  [arg-type]
Found 2 errors in 1 file (checked 330 source files)
```

Reply:
I like this. The more type checking we have in place, the more potential issues we can catch before they blow up.